src/tests/ftest/launch.py

```diff
@@ -809,6 +809,15 @@ def run_tests(test_files, tag_filter, args):
     avocado_logs_dir = os.path.expanduser(avocado_logs_dir[0])
     print("Avocado logs stored in {}".format(avocado_logs_dir))
 
+    # Remove stale job results, if any, before test run
+    stale_dir = "^job-"
+    logs_dir = avocado_logs_dir + '/'
+    if os.path.isdir(logs_dir):
+        sub_dirs = os.listdir(logs_dir)
+        for folder in sub_dirs:
+            if re.search(stale_dir, folder):
```

Comment:
`folder.startswith("job-")` would probably be more efficient than invoking the RE parser.